app/models/course-participation.ts

```diff
@@ -0,0 +1,21 @@
+import Model, { attr, belongsTo } from '@ember-data/model';
+import type CourseModel from './course';
+import type CourseStageModel from './course-stage';
+import type UserModel from './user';
+import type LanguageModel from './language';
+
+export default class CourseParticipationModel extends Model {
+  @belongsTo('course', { async: false, inverse: null }) declare course: CourseModel;
+  @belongsTo('course-stage', { async: false, inverse: null }) declare currentStage: CourseStageModel;
+  @belongsTo('user', { async: false, inverse: 'courseParticipations' }) declare user: UserModel;
+  @belongsTo('language', { async: false, inverse: null }) declare language: LanguageModel;
+
+  @attr('date') declare completedAt: Date;
+  @attr('date') declare lastActivityAt: Date;
+  @attr('date') declare startedAt: Date;
+  @attr('date') declare lastSubmissionAt: Date;
```

Comment:
Same for this - type should be `Date | null`

Reply:
Ignore, this seems to be non-null for all records. Noted this in another comment: 

> Okay looks like all the records we have at the moment have this as non-null. Let's add a database migration in core so that this is codified (and we never create invalid records in the future)